x-pack/plugins/security_solution/public/management/pages/event_filters/view/components/form/index.tsx

```diff
@@ -80,20 +85,47 @@ export const EventFiltersForm: React.FC<EventFiltersFormProps> = memo(
       isGlobal: isGlobalPolicyEffected(exception?.tags),
     });
 
+    const isEditMode = useMemo(() => !!exception?.item_id, [exception?.item_id]);
+    const [wasByPolicy, setWasByPolicy] = useState(!isGlobalPolicyEffected(exception?.tags));
+
+    const showAssignmentSection = useMemo(() => {
+      return (
+        isPlatinumPlus ||
+        (isEditMode &&
+          (!selection.isGlobal || (wasByPolicy && selection.isGlobal && hasFormChanged)))
+      );
+    }, [isEditMode, selection.isGlobal, hasFormChanged, isPlatinumPlus, wasByPolicy]);
+
     // set current policies if not previously selected
     useEffect(() => {
       if (selection.selected.length === 0 && exception?.tags) {
         setSelection(getEffectedPolicySelectionByTags(exception.tags, policies));
       }
     }, [exception?.tags, policies, selection.selected.length]);
 
+    // set initial state of `wasByPolicy` that checks if the initial state of the exception was by policy or not
+    useEffect(() => {
+      if (!hasFormChanged && exception?.tags) {
+        setWasByPolicy(!isGlobalPolicyEffected(exception?.tags));
+      }
+    }, [exception?.tags, hasFormChanged]);
+
     const osOptions: Array<EuiSuperSelectOption<OperatingSystem>> = useMemo(
       () => OPERATING_SYSTEMS.map((os) => ({ value: os, inputDisplay: OS_TITLES[os] })),
       []
     );
 
     const handleOnBuilderChange = useCallback(
       (arg: OnChangeProps) => {
+        if (
```

Comment:
Prevent to dispatch an action when the component is mounted and the exceptionBuilder initialised.

Reply:
This is not fixing a bug but is improving the performance of this form. Before, when the component was mounted, the `eventFiltersChangeForm` action was dispatched for the `ExceptionBuilderComponent` was mounted. This is preventing it to dispatch the action when nothing has changed yet